src/utils/genesysCloudUtils.ts

```diff
@@ -0,0 +1,228 @@
+import { clientConfig } from '../config/clientConfig';
+import moment from 'moment';
+const platformClient = require('purecloud-platform-client-v2/dist/node/purecloud-platform-client-v2.js');
+
+interface IQueue {
+    id: string,
+    activeUsers: number,
+    onQueueUsers: number
+}
+
+const searchApi = new platformClient.SearchApi();
+const usersApi = new platformClient.UsersApi();
+const analyticsApi = new platformClient.AnalyticsApi();
+const tokensApi = new platformClient.TokensApi();
+const routingApi = new platformClient.RoutingApi();
+const presenceApi = new platformClient.PresenceApi();
+// const conversationsApi = new platformClient.ConversationsApi();
+
+/* 
+ * This presence ID is hardcoded because System presence IDs are hardcoded into Genesys Cloud, can never change, and are not unique to orgs or regions
+ * In constrast, Org presences are not hardcoded.
+*/
+const offlinePresenceId = 'ccf3c10a-aa2c-4845-8e8d-f59fa48c58e5';
+
+const client = platformClient.ApiClient.instance;
+const { clientId, redirectUri } = clientConfig;
+
+const cache: any = {};
+
+export function authenticate() {
```

Comment:
This file, specifically,  should be heavily commented. State the purpose of every function and explain every Genesys Cloud API request (briefly). This is the part of the app that people are going to reference to learn how to use our APIs, so being thorough with commenting will help guide people through it.